helper/restore_helper.go

```diff
@@ -389,7 +389,7 @@ func doRestoreAgent() error {
 		logVerbose(fmt.Sprintf("Oid %d, Batch %d: End batch restore", tableOid, batchNum))
 
 		// On resize restore reader might be nil.
-		if !*singleDataFile && !(*isResizeRestore && contentToRestore >= *origSize) {
+		if !*singleDataFile && readers[contentToRestore] != nil {
```

Comment:
Why this happens only on this line? There are some other `if`s that have the same condition, for example, on line 344.

Reply:
there is no bug on line 344, because readers are filled before the main loop on this case (single data file, line 202)

Reply:
Can you modify the comment to tell the reason why a reader could be nil sometimes?

Reply:
> Can you modify the comment to tell the reason why a reader could be nil sometimes?

How about such
```go
// During resize restore, the reader can be nil, for example, if contentToRestore >= *origSize, and also for the next batch after detecting a skip file.
```

Reply:
Sure

Reply:
applied